src/components/alert/bl-alert.ts

```diff
@@ -0,0 +1,139 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { event, EventDispatcher } from '../../utilities/event';
+import style from './bl-alert.css';
+import '../icon/bl-icon';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { iconConverter } from './icon.converter';
+import { ButtonVariant, ButtonKind, ButtonSize } from '../button/bl-button';
+import { classMap } from 'lit/directives/class-map.js';
+
+export type AlertVariant = 'info' | 'warning' | 'success' | 'error';
+
+/**
+ * @tag bl-alert
+ * @summary Baklava Alert component
+ *
+ * @cssproperty --bl-alert-display - Sets the display property of button. Default value is 'inline-block'.
+ */
+
+@customElement('bl-alert')
+export default class BlAlert extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  @property({ reflect: true })
+  variant: AlertVariant = 'info';
+
+  @property()
+  description: 'string';
+
+  @property({ converter: iconConverter() })
+  icon?: boolean | string;
+
+  @property({ type: Boolean })
+  closable = false;
+
+  @property()
+  caption?: string;
+
+  @state()
+  _hidden = false;
+
+  @event('bl-close') private onClose: EventDispatcher<boolean>;
+
+  get _hasAlertCaptionSlot() {
+    return this.querySelector(':scope > [slot="caption"]') !== null;
+  }
+
+  get _getAlertActionSlot() {
+    return this.querySelector(':scope > [slot="action"]');
+  }
+
+  _closeHandler() {
+    this._hidden = true;
+    this.onClose(true);
+  }
+
+  shouldRender(value: undefined | boolean | string, html: TemplateResult) {
+    if (!value) return null;
+    return html;
+  }
+
+  predefinedIcons() {
+    switch (this.variant) {
+      case 'success':
+        return 'check_fill';
+      case 'error':
+        return 'close_fill';
+      default:
+        return this.variant;
+    }
+  }
+
+  getIcon(): string | undefined {
+    if (!this.icon) return;
+    if (typeof this.icon === 'boolean') {
```

Comment:
`this.icon === true` already makes a type check (triple equals) and we want to show predefined icons only if the `icon` is `true`. So instead of checking the type, just checking the value with triple equals is more efficient.

Reply:
oh, now i got it. Since the converter handles it, we don't have to check it here. I'm assuming that after i've added the converter, i've just forgot to refactore here. Thanks!